### ISSUE TITLE
LearnedPositionalEmbedding: precompute positions and use a padding_idx

### ISSUE THREAD

**Issue by villmow** [pytorch/fairseq#1497]
Hi,

I want to feed precomputed positions into my learned positional embedding, but if I want to do that this code prohibits me from having a `padding_idx`. I know my padding index and I want to be able to use it in my precomputed positions.

https://github.com/pytorch/fairseq/blob/0e2e5f1fc475c2b87f19c183f65351e13e38ffde/fairseq/modules/learned_positional_embedding.py#L28-L35

Can you explain the intuition why you prohibit having a pad_idx in that case?

Thanks in advance!

**Comment by myleott** [comment]
We don't usually pass in `positions`, this was added by @kartikayk for another use case.

Note that when positions is not None, this module is equivalent to nn.Embedding, so just use that.
